packages/devtools/src/schema.ts

```diff
@@ -0,0 +1,415 @@
+  if (!(fieldType instanceof GraphQLObjectType)) {
+    return
+  }
+  return {
+    $ref: `#/$defs/${fieldType.name}`,
+  }
+}
+
+/** @internal */
+function arrayFieldSchemaFromFieldDefinition(
+  fieldName: string,
+  fieldType: GraphQLOutputType,
+  ceramicExtensions?: CeramicGraphQLTypeExtensions
+): Record<string, any> | undefined {
+  if (!(fieldType instanceof GraphQLList)) {
+    return
+  }
+
+  const result: JSONSchema.Array = {
+    type: 'array',
+    items: fieldSchemaFromFieldDefinition(
+      fieldName,
+      fieldType.ofType,
+      ceramicExtensions // ceramicExtensions are applied recursively to array items
+    ),
+  }
+
+  if (ceramicExtensions) {
+    if (ceramicExtensions.arrayLength !== undefined) {
+      if (ceramicExtensions.arrayLength.min) {
+        result.minItems = ceramicExtensions.arrayLength.min
+      }
+      if (ceramicExtensions.arrayLength.max) {
+        result.maxItems = ceramicExtensions.arrayLength.max
+      }
+    }
+  }
+  return result
+}
+
+/** @internal */
+function defaultFieldSchemaFromFieldDefinition(
+  fieldType: GraphQLOutputType,
+  ceramicExtensions?: CeramicGraphQLTypeExtensions
+): Record<string, any> {
+  let result: Record<string, any> = {
+    type: fieldType.toString().toLowerCase(),
+  }
+
+  if (fieldTypeIsinstanceOfOrWraps(fieldType, GraphQLDID)) {
+    result = {
+      ...result,
+      type: 'string',
+      title: GraphQLDID.toString(),
+      pattern: "/^did:[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+:[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+$/",
+      maxLength: 80,
+    }
+  }
+
+  if (fieldTypeIsinstanceOfOrWraps(fieldType, GraphQLStreamReference)) {
+    result = {
+      ...result,
+      type: 'string',
+      title: 'StreamReference',
+      pattern: '<TBD>', //FIXME: define the pattern for StreamReference strings
+      maxLength: 80,
+    }
+  }
+
+  if (fieldTypeIsinstanceOfOrWraps(fieldType, GraphQLID)) {
+    result = {
+      ...result,
+      type: 'string',
+      title: 'GraphQLID', // TODO: Should we just use GraphQLID.toString() here, which equals to "ID"?
+    }
+  }
+
+  if (fieldTypeIsinstanceOfOrWraps(fieldType, GraphQLInt)) {
+    result = {
+      ...result,
+      type: 'integer',
+    }
+  }
+
+  if (fieldTypeIsinstanceOfOrWraps(fieldType, GraphQLFloat)) {
+    result = {
+      ...result,
+      type: 'number',
+    }
+  }
+
+  if (fieldTypeIsinstanceOfOrWraps(fieldType, GraphQLBoolean)) {
+    result = {
+      ...result,
+      type: 'boolean',
+    }
+  }
+
+  if (fieldTypeIsinstanceOfOrWraps(fieldType, GraphQLString)) {
+    result = {
+      ...result,
+      type: 'string',
+    }
+  }
+
+  if (ceramicExtensions) {
+    if (ceramicExtensions.length !== undefined) {
+      result = {
+        ...result,
+        type: 'string',
+        maxLength: ceramicExtensions.length.max,
+        minLength: ceramicExtensions.length.min,
+      }
+    }
+
+    if (ceramicExtensions.intRange !== undefined) {
+      result = {
+        ...result,
+        type: 'integer',
```

Comment:
I'm overriding the type here, because according to our spec, we expect 'integer', but GraphQL returns 'int'. The validation is done earlier in `compositeDirectivesTransformer(...)`